examples/simple-sampler/stdout_exporter.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include <iostream>
+
+namespace trace    = opentelemetry::trace;
+namespace nostd    = opentelemetry::nostd;
+namespace sdktrace = opentelemetry::sdk::trace;
+
+class StdoutExporter final : public sdktrace::SpanExporter
+{
+  std::unique_ptr<sdktrace::Recordable> MakeRecordable() noexcept
+  {
+    return std::unique_ptr<sdktrace::Recordable>(new sdktrace::SpanData);
+  }
+
+  sdktrace::ExportResult Export(
+      const nostd::span<std::unique_ptr<sdktrace::Recordable>> &spans) noexcept
+  {
+    for (auto &recordable : spans)
+    {
+      auto span = std::unique_ptr<sdktrace::SpanData>(
+          static_cast<sdktrace::SpanData *>(recordable.release()));
```

Comment:
Hm, this takes ownership of the recordable object. This is fairly dangerous, since the param "spans" will no longer have ownership of its Recordable members. Note that unique_ptr.release() is not a const method. I believe the compiler allows this because you're not mutating the input param "spans", just the container objects (this is just my theory though).

Instead of converting recordable into another unique_ptr, consider just casting the pointer itself, ie.

const auto* span = static_cast\<const sdktrace::SpanData*\>(recordable.get())